doozer/doozerlib/metadata.py

```diff
@@ -401,11 +402,121 @@ async def get_latest_build(self, **kwargs):
             return await self.get_latest_brew_build_async(**kwargs)
 
         elif self.runtime.build_system == 'konflux':
-            raise NotImplementedError
+            return await self.get_latest_konflux_build(**kwargs)
 
         else:
             raise ValueError(f'Invalid value for --build-system: {self.runtime.build_system}')
 
+    async def get_pinned_konflux_build(self, el_target: int):
+        """
+        Return the assembly pinned NVR.
+        Under 'is' for RPMs, we expect 'el7' and/or 'el8', etc. For images, just 'nvr'.
+        """
+
+        is_config = self.config['is']
+
+        if self.meta_type == 'rpm':
+            if el_target is None:
+                raise ValueError(
+                    f'Expected el_target to be set when querying a pinned RPM component {self.distgit_key}')
+            is_nvr = is_config[f'el{el_target}']
+            if not is_nvr:
+                self.logger.warning('No pinned NVR found for RPM %s and target %s', self.rpm_name, el_target)
```

Comment:
For consistency with line 436, an error should be raised instead of returning none.

Reply:
I followed the logic we already had [here](https://github.com/openshift-eng/art-tools/blob/main/doozer/doozerlib/metadata.py#L552). It looks like for RPMs `get_latest_brew_build` isn't throwing an exception if a default != -1 is set (which is always the case, we pass in default=None everywhere but [here](https://github.com/openshift-eng/art-tools/blob/main/doozer/doozerlib/cli/release_gen_assembly.py#L477))

Reply:
default_return (https://github.com/openshift-eng/art-tools/blob/27708d09b004e12ab590f95ca5e42271cadb7475/doozer/doozerlib/metadata.py#L485) raises IOError when default == -1.

Reply:
yeah so for images we always pass in defautl = None so an exception is never thrown if a container build that was pinned isn't found. If you feel that should change, we should probably change it in legacy code as well